src/Microsoft.Kusto.ServiceLayer/DataSource/KustoClient.cs

```diff
@@ -195,20 +195,16 @@ public async Task ExecuteControlCommandAsync(string command, bool throwOnError,
         /// Executes a query.
         /// </summary>
         /// <param name="query">The query.</param>
+        /// <param name="cancellationToken"></param>
+        /// <param name="databaseName"></param>
         /// <returns>The results.</returns>
-        public async Task<IEnumerable<T>> ExecuteQueryAsync<T>(string query, CancellationToken cancellationToken, string databaseName = null)
+        public async Task<IEnumerable<T>> ExecuteQueryAsync<T>(string query, CancellationToken cancellationToken,
+            string databaseName = null)
         {
-            try
-            {
-                var resultReader = ExecuteQuery(query, cancellationToken, databaseName);
-                var results = KustoDataReaderParser.ParseV1(resultReader, null);
-                var tableReader = results[WellKnownDataSet.PrimaryResult].Single().TableData.CreateDataReader();
-                return await Task.FromResult(new ObjectReader<T>(tableReader));
-            }
-            catch (Exception)
```

Comment:
Removed catch block so exceptions will be bubbled up and returned to the ADS client.

Reply:
Do we have retries in the ExecuteQuery path. That will make this code path resilient to intermittent network/service issues.

Reply:
We have a single retry happening inside the ExecuteQuery function.